ci-configs/.gitlab-ci.yml

```diff
@@ -0,0 +1,67 @@
+image: node:8
+
+# Cache dependencies.
+cache:
+  paths:
+    - node_modules/
+
+# Define the desired stages.
+test:
+  stage: test
+  script:
+    - yarn install
+    - yarn test
+
+lint:
+  stage: test
+  script:
+    - yarn install
+    - yarn lint
+
+# Code quality setup.
+code_quality:
+  stage: test
+  image: docker:stable
+  variables:
+    DOCKER_DRIVER: overlay2
+  allow_failure: true
+  services:
+    - docker:stable-dind
+  script:
+    - export SP_VERSION=$(echo "$CI_SERVER_VERSION" | sed 's/^\([0-9]*\)\.\([0-9]*\).*/\1-\2-stable/')
+    - docker run
+      --env SOURCE_CODE="$PWD"
+      --volume "$PWD":/code
+      --volume /var/run/docker.sock:/var/run/docker.sock
+      "registry.gitlab.com/gitlab-org/security-products/codequality:$SP_VERSION" /code
+  artifacts:
+    reports:
+      codequality: gl-code-quality-report.json
+
+# Auto-deployment to GitLab pages.
+pages:
+  stage: deploy
+  script:
```

Comment:
Let's make it so that this step exits with an info log when `$DEPLOY_TO_GL_PAGES` is not set. So it can be always run safely.